aldryn_client/cloud.py

```diff
@@ -85,6 +89,19 @@ def upload_boilerplate(self, archive_obj):
         )
         return request.request()
 
+    def download_db(self, website_slug):
+        request = api_requests.DownloadDBRequest(
+            self.session,
+            url_kwargs={'website_slug': website_slug},
+        )
+        temp_dir = create_temp_dir()
+        response = request.request()
+        with tarfile.open(
+                fileobj=StringIO(response.content), mode='r:gz'
```

Comment:
you should also be able to pipe it into pg_restore directly `tar -xzOf dump.tar.gz | pg_restore ...`, but that requires it to be run through `sh -c '...'`.